modules/4337/contracts/experimental/verifiers/WebAuthnVerifier.sol

```diff
@@ -102,7 +104,61 @@ contract WebAuthnVerifier is IWebAuthnVerifier, P256Wrapper {
         bytes32 challenge,
         bytes calldata clientDataFields
     ) internal pure returns (bytes32 message) {
-        string memory encodedChallenge = Base64Url.encode(abi.encodePacked(challenge));
+
+        // Result is 44 bytes long because the encoded challenge is 32 bytes long
+        // 4*(32 + 2)/3 = 44 after rounding.
+        string memory table = ENCODING_TABLE;
+        string memory encodedChallenge = new string(44);
```

Comment:
Instead of a `new string(44)`, you can actually decode this directly into `clientDataJson` by building it with:

```solidity
        bytes memory clientDataJson = abi.encodePacked(
            '{"type":"webauthn.get","challenge":"AAAAAAAA....",',
            clientDataFields,
            "}"
        );
```

Reply:
This avoids the copy loop from `abi.encodePacked`-ing a dynamic string.

Reply:
done in https://github.com/safe-global/safe-modules/pull/318/commits/f4c2e88d1d7c757b63ad85cc89b93f1e70a0a6f5